File: Task102/app/src/main/java/com/example/task102/MainActivity.java
package com.example.task102;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TextView textView = findViewById(R.id.textView);
        textView.setTextColor(Color.RED);
        Button button = findViewById(R.id.button);
        Button buttonNavigate = findViewById(R.id.buttonNavigate);

        button.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                textView.setText("Button Clicked!");
                //Show a Toast message
                Toast.makeText(MainActivity.this, "Button was clicked!", Toast.LENGTH_SHORT).show();
            }
        });
        buttonNavigate.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this,
                        SecondActivity.class);
                intent.putExtra("EXTRA_MESSAGE", "Hello from MainActivity!");
                startActivity(intent);


            }
        });

    }
}

File: Task1Kotlin/app/src/main/java/com/example/task1kotlin/MainActivity.kt
package com.example.task1kotlin

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.task1kotlin.R

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val textView = findViewById<TextView>(R.id.textView)
        textView.setTextColor(Color.RED)
        val button = findViewById<Button>(R.id.button)
        val buttonNavigate = findViewById<Button>(R.id.buttonNavigate)

        button.setOnClickListener {
            textView.text = "Button Clicked!"
            //Show a Toast message
            Toast.makeText(
                this@MainActivity,
                "Button was clicked!",
                Toast.LENGTH_SHORT
            ).show()
        }
        buttonNavigate.setOnClickListener {
            val intent = Intent(
                this@MainActivity,
                SecondActivity::class.java
            )
            intent.putExtra("EXTRA_MESSAGE", "Hello from MainActivity!")
            startActivity(intent)
        }
    }
}